ray-operator/controllers/ray/rayjob_controller_scheduled_test.go

```diff
@@ -0,0 +1,211 @@
+		rayJob.Spec.Schedule = cronSchedule
+		rayJob.Spec.ShutdownAfterJobFinishes = false
+		rayCluster := &rayv1.RayCluster{}
+
+		It("Verify RayJob spec", func() {
+			Expect(rayJob.Spec.ShutdownAfterJobFinishes).To(BeFalse())
+			Expect(rayJob.Spec.Schedule).To(Not(BeEmpty()))
+		})
+
+		It("should create a RayJob object with the schedule", func() {
+			err := k8sClient.Create(ctx, rayJob)
+			Expect(err).NotTo(HaveOccurred(), "failed to create test scheduled RayJob resource")
+		})
+
+		It("should start in the Scheduled state", func() {
+			Eventually(
+				getRayJobDeploymentStatus(ctx, rayJob),
+				time.Second*60, time.Microsecond*500).Should(Equal(rayv1.JobDeploymentStatusScheduled),
+				"JobDeploymentStatus should be Initializing")
+		})
+
+		It("should NOT create a raycluster object while scheduled", func() {
+			Consistently(
+				getRayClusterNameForRayJob(ctx, rayJob),
+				time.Second*3, time.Millisecond*500).Should(BeEmpty())
+		})
+
+		It("should update schedule string in its spec", func() {
+			rayJobLookupKey := types.NamespacedName{Name: rayJob.Name, Namespace: rayJob.Namespace}
+			fetchedRayJob := &rayv1.RayJob{}
+			newSchedule := "*/1 * * * *"
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, rayJobLookupKey, fetchedRayJob)
+				return err == nil
+			}, time.Second*5, time.Millisecond*200).Should(BeTrue(), "expected to fetch the RayJob before update")
+
+			fetchedRayJob.Spec.Schedule = newSchedule
+
+			err := updateRayJobScheduleField(ctx, fetchedRayJob, newSchedule)
+			Expect(err).NotTo(HaveOccurred(), "failed to update RayJob's schedule in spec")
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, rayJobLookupKey, fetchedRayJob)
+				if err != nil {
+					return false
+				}
+				return fetchedRayJob.Spec.Schedule == newSchedule
+			}, time.Second*10, time.Millisecond*500).Should(BeTrue(), "expected RayJob's Spec.Schedule to be updated and match")
+		})
+
+		// The cron job runs every minute so it will take at most 1 minute to run
+		It("should transition to the Initializing", func() {
+			Eventually(
+				getRayJobDeploymentStatus(ctx, rayJob),
+				time.Second*70, time.Microsecond*500).Should(Equal(rayv1.JobDeploymentStatusInitializing),
+				"JobDeploymentStatus should be Initializing")
+		})
+
+		It("should create a raycluster object", func() {
+			Eventually(
+				getRayClusterNameForRayJob(ctx, rayJob),
+				time.Second*15, time.Millisecond*500).Should(Not(BeEmpty()))
+			Eventually(
+
+				getResourceFunc(ctx, common.RayJobRayClusterNamespacedName(rayJob), rayCluster),
+				time.Second*3, time.Millisecond*500).Should(Succeed())
+		})
+
+		// We are checking if LastScheduleTime is correctly set
+		It("should have LastScheduleTime updated in its status", func() {
+			rayJobLookupKey := types.NamespacedName{Name: rayJob.Name, Namespace: rayJob.Namespace}
+			fetchedRayJob := &rayv1.RayJob{}
+
+			var lastScheduleTime *time.Time
+			Eventually(func() bool {
```

Comment:
Please use this pattern to fetch the last scheduled time https://github.com/ray-project/kuberay/blob/master/ray-operator/controllers/ray/rayjob_controller_test.go#L376-L378